From 2961cf75427102723730153ff5eb93b850db9b14 Mon Sep 17 00:00:00 2001
From: Shuguang Gong <shuguang.gong@intel.com>
Date: Mon, 2 Feb 2015 10:57:36 +0800
Subject: [PATCH 247/255] ov2685: fix wrong ctrl_id and descriptions

Change-Id: Id23cc1d9b7911baa71a884e56f5fd23afec2c0be
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-19920
Signed-off-by: Shuguang Gong <shuguang.gong@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/38876
Approver: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
---
 .../external_drivers/camera/drivers/media/i2c/ov2685.c | 18 ++++++++++++++----
 1 file changed, 14 insertions(+), 4 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov2685.c b/drivers/external_drivers/camera/drivers/media/i2c/ov2685.c
index 7314c1f..7592cf1 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov2685.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov2685.c
@@ -657,7 +657,7 @@ static int ov2685_s_ctrl(struct v4l2_ctrl *ctrl)
 
 	switch (ctrl->id) {
 	case V4L2_CID_EXPOSURE:
-		dev_dbg(&client->dev, "%s: CID_VFLIP:%d.\n",
+		dev_dbg(&client->dev, "%s: CID_EXPOSURE:%d.\n",
 			__func__, ctrl->val);
 		ret = ov2685_s_exposure(&dev->sd, ctrl->val);
 		break;
@@ -672,17 +672,17 @@ static int ov2685_s_ctrl(struct v4l2_ctrl *ctrl)
 		ret = ov2685_t_hflip(&dev->sd, ctrl->val);
 		break;
 	case V4L2_CID_POWER_LINE_FREQUENCY:
-		dev_dbg(&client->dev, "%s: CID_HFLIP:%d.\n",
+		dev_dbg(&client->dev, "%s: CID_POWER_LINE_FREQUENCY:%d.\n",
 			__func__, ctrl->val);
 		ret = ov2685_s_freq(&dev->sd, ctrl->val);
 		break;
 	case V4L2_CID_AUTO_N_PRESET_WHITE_BALANCE:
-		dev_dbg(&client->dev, "%s: CID_HFLIP:%d.\n",
+		dev_dbg(&client->dev, "%s: CID_WHITE_BALANCE:%d.\n",
 			__func__, ctrl->val);
 		ret = ov2685_s_wb(&dev->sd, ctrl->val);
 		break;
 	case V4L2_CID_SCENE_MODE:
-		dev_dbg(&client->dev, "%s: CID_HFLIP:%d.\n",
+		dev_dbg(&client->dev, "%s: CID_SCENE_MODE:%d.\n",
 			__func__, ctrl->val);
 		ret = ov2685_s_scene(&dev->sd, ctrl->val);
 		break;
@@ -728,6 +728,16 @@ static const struct v4l2_ctrl_ops ctrl_ops = {
 static const struct v4l2_ctrl_config ov2685_controls[] = {
 	{
 		.ops = &ctrl_ops,
+		.id = V4L2_CID_EXPOSURE,
+		.type = V4L2_CTRL_TYPE_INTEGER,
+		.name = "exposure biasx",
+		.min = -2,
+		.max = 2,
+		.step = 0x01,
+		.def = 0x00,
+	},
+	{
+		.ops = &ctrl_ops,
 		.id = V4L2_CID_EXPOSURE_ABSOLUTE,
 		.type = V4L2_CTRL_TYPE_INTEGER,
 		.name = "exposure",
-- 
1.9.1

